src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -86,66 +95,74 @@ public class RD2_OpportunityEvaluationService {
 
     /**
      * @description Pass the set of Recurring Donation Id's to be processed
-     * @param rdIds
+     * @param rdIds Recurring Donation Ids
+     * @return RD2_OpportunityEvaluationService The service instance
      */
     public RD2_OpportunityEvaluationService withRds(Set<Id> rdIds) {
         this.rdIds = rdIds;
         return this;
     }
 
+    /**
+     * @description Returns true if any of the key fields used to create or manage installment
+     * Opportunities has been changed.
+     * @param rd Changed Recurring Donation
+     * @param oldRd Old Recurring Donation
+     * @return Boolean
+     */
+    public Boolean hasKeyFieldChanged(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {
+        return 
+            rd.npe03__Contact__c != oldRd.npe03__Contact__c ||
+            rd.npe03__Organization__c != oldRd.npe03__Organization__c ||
+            rd.npe03__Amount__c != oldRd.npe03__Amount__c ||
+            rd.npe03__Installment_Period__c != oldRd.npe03__Installment_Period__c ||
+            rd.InstallmentFrequency__c != oldRd.InstallmentFrequency__c ||
+            rd.StartDate__c != oldRd.StartDate__c ||
+            rd.Day_of_Month__c != oldRd.Day_of_Month__c ||
+            rd.Status__c != oldRd.Status__c;
+    }
+
     /**
      * @description Primary processor
+     * @return RD2_OpportunityEvaluationService The service instance
      */
     public RD2_OpportunityEvaluationService evaluateOpportunities() {
 
-        Map<npe03__Recurring_Donation__c, List<Date>> rdsWithCloseDates = new Map<npe03__Recurring_Donation__c, List<Date>>();
+        Map<npe03__Recurring_Donation__c, List<Date>> newRDsWithCloseDates = new Map<npe03__Recurring_Donation__c, List<Date>>();
+        List<Opportunity> existingOpenOpps = new List<Opportunity>();
 
         for (npe03__Recurring_Donation__c rd : rds) {
-            if (!isCandidate(rd)) {
-                continue;
-            }
 
             if (isNewRecurringDonation(rd)) {
-                rdsWithCloseDates.put(rd, getInstallmentOppCloseDates(rd) );
+                newRDsWithCloseDates.put(rd, getInstallmentOppCloseDates(rd) );
                 continue;
             }
-            // TODO Update the Future Installment Opportunity if needed
+            
+            if (rd.Status__c == RD2_Constants.STATUS_CLOSED) {
```

Comment:
@force2b , when RD data is migrated and `Status__c` updated to Closed for specific RDs, the open Opportunities  will be deleted/closed/unchanged based on the RD settings. If our requirements are to leave Opportunities for migrated RD currently, should we set RD settings to no action during data migration?